src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/DES/DESCipherTests.cs

```diff
@@ -146,5 +146,45 @@ public static void DesRoundTrip(CipherMode cipherMode, PaddingMode paddingMode,
                 Assert.Equal<byte>(expectedDecryptedBytes, decrypted);
             }
         }
+
+        public static void DesReuseEncryptorDecryptor()
+        {
+            using (DES alg = DESFactory.Create())
+            {
+                alg.Key = s_randomKey_64.HexToByteArray();
+                alg.IV = s_randomIv_64.HexToByteArray();
+                alg.Padding = PaddingMode.PKCS7;
+                alg.Mode = CipherMode.CBC;
+
+                using (ICryptoTransform encryptor = alg.CreateEncryptor())
+                using (ICryptoTransform decryptor = alg.CreateDecryptor())
+                {
+                    for (int i = 0; i < 2; i++)
+                    {
+                        byte[] plainText1 = s_multiBlockString.HexToByteArray();
+                        byte[] cipher1 = encryptor.Transform(plainText1);
+                        byte[] expectedCipher1 = (
+                            "7264319AE3C504148CD4A19B4FDC7D2ACCCB0A08D60CBE2B885DCB2C1A86ED9CA51006E33859B03E00F5B57801EFF745" +
+                            "F7A577842461CF39AC143505EC326233E66343A46FEADE9E8456D8AC6A84A1C32E6792857F062400EA9053D17AD3C35D").HexToByteArray();
+                        Assert.Equal<byte>(expectedCipher1, cipher1);
+
+                        byte[] decrypted1 = decryptor.Transform(cipher1);
+                        byte[] expectedDecrypted1 = s_multiBlockString.HexToByteArray();
+                        Assert.Equal<byte>(expectedDecrypted1, decrypted1);
+
+                        byte[] plainText2 = s_multiBlockString_8.HexToByteArray();
```

Comment:
The pattern was copied from 3des. It does iterate. Note that there was an issue with second decrypt (fixed now) so added a blank call to encrypt to ensure final=true clears the handle and resets the IV